### PR TITLE
Init the global variables once per host, not each time for every site

### PR DESCRIPTION
**Pull request by jxchinaren**

### CODE REVIEW

File: src/ee/common/executorcontext.cpp [hunk 1]
```diff
@@ -32,6 +35,33 @@ static pthread_once_t static_keyOnce = PTHREAD_ONCE_INIT;
 
 static void createThreadLocalKey() {
     (void)pthread_key_create( &static_key, NULL);
+
```
Comment (wolffcm) [inline src/ee/common/executorcontext.cpp]: This seems like a good place to put stuff that has to happen exactly once for the whole process, but maybe the name of this method should be changed to "processWideInit" or something?

Also, pthread_key_create might call malloc under the hood, do we want call it after we've made the calls to mallopt?

**Comment by wolffcm** [comment]
That's all my comments.

**Comment by jxchinaren** [comment]
@wolffcm I just resolved your comments. Thanks.

**Comment by wolffcm** [comment]
Looks good to me.

**Comment by jxchinaren** [comment]
I will close this pull request to squash merge with one single commit, it's better to cherry pick and patch to previous version for Ruth.

**Pull request closed without merging**